### DISCUSSION TITLE
Custom Architecture Kits

### DISCUSSION THREAD

**Discussion by joone-org** [mozilla/hubs#3611] [Hubs Cloud Questions] [answered]
Is there a way to create custom kits, similar to the architecture and rock kits, using my own set of models (in Hubs Cloud)?

**Comment by Utopiah** [comment]
I'd have a look at [ui/assets/KitSource.js](https://github.com/mozilla/Spoke/blob/master/src/ui/assets/KitSource.js) and [hubs-blender-exporter](https://github.com/MozillaReality/hubs-blender-exporter) as we can see `node.extensions.MOZ_hubs_components["kit-piece"];` we can assume that the exporter is used to add the component to all children of the roof node of the linked gltf file.

In fact comparing [ui/assets/sources/ArchitectureKitSource.js](https://github.com/mozilla/Spoke/blob/master/src/ui/assets/sources/ArchitectureKitSource.js) with [ui/assets/sources/RockKitSource.js](https://github.com/mozilla/Spoke/blob/master/src/ui/assets/sources/RockKitSource.js) you can see it all seems to be packed (with https://hubs.mozilla.com/spoke/kits/package ) in the glb with some meta-data provided.

**Comment by j-conrad** [comment, marked as answer]
Yes... almost.

The workflow on the Blender side exists, and as @Utopiah pointed out, you can see where it lives in the exporter code. However, the final steps in getting a kit into Spoke currently requires a change to the Spoke code itself. What we still need to do (and it's definitely on the to-do list) is to create a workflow so that users can more easily surface their kit in Spoke--locally or otherwise. Of course, we'll also have to make the appropriate documentation.
So far, I made the initial 'Architecture Kit' and 'Rock Kit', but there's so much potential for making many more, especially with all the content available on the web, so your desire to make them is noted and appreciated. Hopefully we can get to this in the near future.
